### PR TITLE
Update zig `@panic` calls to `roc_panic` for numeric errors

### LINKED ISSUE #6027: Num.div by 0 on Dec should crash, but doesn't
To reproduce, in the repl:

```
» 1/0

  1000000000000000000 : Frac *
```

Expected behavior: a crash report on division by zero, like you get for integer divide by zero.

(This can't return something like infinity or NaN because it's a `Dec`, not a float, and `Dec`'s fixed-point representation is based on integers; it doesn't have an equivalent concept.)

### PR DESCRIPTION
**Pull request by JRMurr**
fixes #6027 

fyi I dont really know zig so please let me know if i did something wrong

## The issue

calls to `@panic` in the zig bytecode generator would not bubble up the rest of the compiler so we would get nonsene numbers back instead of raise the error to the roc code that should have errored. For example ` 1dec / 0` would give a giant number in wasm and 0 in llvm

## What I did

updated all the numeric related calls to `@panic` to `roc_panic`. TBH not sure on the difference between the two but added some tests and they only fail when expected when using `roc_panic`

## Some errors I'm seeing

I posted some info in [zulip](https://roc.zulipchat.com/#narrow/stream/316715-contributing/topic/zig.20.40panic.20vs.20roc_panic/near/403512756)

TLDR we still show "normal" output when a `panic` or call to `crash` happens. 

I could see the argument that its nice to display the type signature still but maybe we hid the value for crashes?

## Misc changes

added node and zls to the dev shell since npx is needed for the site dev and zls is nice when editing zig.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -116,6 +116,8 @@
           wasm-pack # for repl_wasm
           jq # used in several bash scripts
           cargo-nextest # used to give more info for segfaults for gen tests
+          nodejs_18 # for npx in `build-dev-local`
+          zls # zig language server
```
Comment (folkertdev) [inline flake.nix:120]: @Anton-4 any concerns about including these?

**Comment by JRMurr** [comment]
looking into failures. The apple-silicon failure looks like https://github.com/roc-lang/roc/issues/5932 so possibly a flake?

The valgrind error i'm having trouble reproducing locally. will investigate

File: flake.nix (hunk 1 above)
Comment (Anton-4) [reply flake.nix:120]: Yes, I think we should ditch npx and use [simple-http-server](https://crates.io/crates/simple-http-server) like before. I'll check on zulip if that's ok.

Including zls seems pretty reasonable.

Thanks for alerting me Folkert :)

Comment (Anton-4) [reply flake.nix:120]: > I think we should ditch npx and use [simple-http-server](https://crates.io/crates/simple-http-server) like before. I'll check on zulip if that's ok.

Richard said it's ok, he was having some performance problem with the feedback loop working on wasm stuff with the simple-http-server. But he's fine with going back to simple-http-server.

Comment (Anton-4) [reply flake.nix:120]: We should do that in separate PR though

Comment (JRMurr) [reply flake.nix:120]: sounds good, ill remove node and keep zls

File: crates/compiler/builtins/bitcode/src/panic.zig [hunk 2]
```diff
@@ -2,14 +2,14 @@ const std = @import("std");
...
+pub fn panic_help(msg: []const u8, tag_id: u32) noreturn {
     var str = RocStr.init(msg.ptr, msg.len);
     roc_panic(&str, tag_id);
 }
 
 // must export this explicitly because right now it is not used from zig code
-pub fn panic(msg: *const RocStr, alignment: u32) callconv(.C) void {
+pub fn panic(msg: *const RocStr, alignment: u32) callconv(.C) noreturn {
     return roc_panic(msg, alignment);
 }
```
Comment (JRMurr) [inline crates/compiler/builtins/bitcode/src/panic.zig:15]: I was having a weird error in CI were valgrind was crashing but was not happening locally

My gut feeling is that us adding `unreachable` after a `roc_panic` call is whats causing the issue. 

Looks like unreachable is undefined behavior when compiled with RelaseFast https://ziglang.org/documentation/master/#try which i think we do

This return type lets you do
```
if (some_cond) {
    roc_panic(...)
} else {
    return foo
}
```

without zig type errors

File: crates/compiler/builtins/bitcode/src/dec.zig [hunk 3]
```diff
@@ -632,7 +629,7 @@ fn mul_and_decimalize(a: u128, b: u128) i128 {
     const d = answer[0];
 
     if (overflowed == 1) {
-        @panic("TODO runtime exception for overflow!");
+        roc_panic("Decimal multiplication overflow22!", 0);
```
Comment (bhansconnect) [inline crates/compiler/builtins/bitcode/src/dec.zig:632]:
```suggestion
        roc_panic("Decimal multiplication overflow!", 0);
```

File: crates/compiler/builtins/bitcode/src/dec.zig [hunk 4]
```diff
@@ -1208,7 +1205,7 @@ pub fn fromF64C(arg: f64) callconv(.C) i128 {
     if (@call(.always_inline, RocDec.fromF64, .{arg})) |dec| {
         return dec.num;
     } else {
-        @panic("TODO runtime exception failing convert f64 to RocDec");
+        roc_panic("Decimal conversion from f64!", 0);
```
Comment (bhansconnect) [inline crates/compiler/builtins/bitcode/src/dec.zig:1208]:
```suggestion
        roc_panic("Decimal conversion from f64 failed!", 0);
```

File: crates/compiler/builtins/bitcode/src/dec.zig [hunk 5]
```diff
@@ -1217,7 +1214,7 @@ pub fn fromF32C(arg_f32: f32) callconv(.C) i128 {
     if (@call(.always_inline, RocDec.fromF64, .{arg_f64})) |dec| {
         return dec.num;
     } else {
-        @panic("TODO runtime exception failing convert f64 to RocDec");
+        roc_panic("Decimal conversion from f32!", 0);
```
Comment (bhansconnect) [inline crates/compiler/builtins/bitcode/src/dec.zig:1217]:
```suggestion
        roc_panic("Decimal conversion from f32 failed!", 0);
```

File: crates/compiler/builtins/bitcode/src/dec.zig [hunk 6]
```diff
@@ -1232,7 +1229,7 @@ pub fn exportFromInt(comptime T: type, comptime name: []const u8) void {
 
             const answer = @mulWithOverflow(this, RocDec.one_point_zero_i128);
             if (answer[1] == 1) {
-                @panic("TODO runtime exception failing convert integer to RocDec");
+                roc_panic("Decimal conversion from integer!", 0);
```
Comment (bhansconnect) [inline crates/compiler/builtins/bitcode/src/dec.zig:1232]:
```suggestion
                roc_panic("Decimal conversion from integer failed!", 0);
```

File: crates/compiler/builtins/bitcode/src/num.zig [hunk 7]
```diff
@@ -233,7 +233,9 @@ pub fn exportCeiling(comptime F: type, comptime T: type, comptime name: []const
 pub fn exportDivCeil(comptime T: type, comptime name: []const u8) void {
     comptime var f = struct {
         fn func(a: T, b: T) callconv(.C) T {
-            return math.divCeil(T, a, b) catch @panic("TODO runtime exception for dividing by 0!");
+            return math.divCeil(T, a, b) catch {
+                roc_panic("integer division by 0!", 0);
```
Comment (bhansconnect) [inline crates/compiler/builtins/bitcode/src/num.zig:237]: Would be worth trying to convert these all back to `Integer`. Probably will require update some tests. I assume most in `cargo test-gen-llvm`

File: crates/compiler/builtins/bitcode/src/panic.zig (hunk 2 above)
Comment (bhansconnect) [reply crates/compiler/builtins/bitcode/src/panic.zig:15]: So glad this worked. Thanks for the find

File: crates/lang_srv/debug_server.sh [hunk 8]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
```
Comment (bhansconnect) [inline crates/lang_srv/debug_server.sh:1]: It looks like you have a number of unrelated changes in this file and below. Probably don't want those here?

File: crates/compiler/builtins/bitcode/src/num.zig (hunk 7 above)
Comment (JRMurr) [reply crates/compiler/builtins/bitcode/src/num.zig:237]: Ill give that a shot

File: crates/lang_srv/debug_server.sh (hunk 8 above)
Comment (JRMurr) [reply crates/lang_srv/debug_server.sh:1]: ahh yea this file is not related to this branch. I can remove

the readme/gitignore updates were useful when i was testing the wasm repl so i figured those were fine to leave in but can move out

File: crates/compiler/builtins/bitcode/src/num.zig (hunk 7 above)
Comment (JRMurr) [reply crates/compiler/builtins/bitcode/src/num.zig]: Did a punch of find replace and made all the panics i found capitalize `Integer`

File: crates/repl_wasm/README.md
Comment (bhansconnect) [inline crates/repl_wasm/README.md:1]: Still a bit confused about the random repl and flake changes, but otherwise, this pr looks great.

Comment (JRMurr) [reply crates/repl_wasm/README.md:1]: The docs here were out of date, linking to the public dir did not seem to hot reload for me

**Comment by bhansconnect** [comment]
Do to some ci testing changes, I think this needs rebased or have main merged then tests should run fine and it should be good to merge.

**Comment by JRMurr** [comment]
@bhansconnect you ok with merging this with the doc changes or do you want me to pull those out?

**Review by bhansconnect — APPROVED** [review]
